### PR TITLE
Printing cleanup

### PR DESCRIPTION
**Pull request by diyessi**
Add Input/Output
Simplify element type
Simplfy Node printing  implementation, try to de-noise it a bit, add op version
Enable printing of Node*

Samples:
`v0::Add Add_2(Parameter_0[0]:f32{2,3}, Parameter_1[0]:f32{2,3}):(f32{2,3})`
`Parameter_0[0]:f32{2,3}`
`Add_2.input(1):f32{2,3}`

### CODE REVIEW

File: doc/sphinx/source/inspection/debug_core.rst [hunk 1]
```diff
@@ -140,4 +140,11 @@ in the unit test for this feature.
...
-.. _precompiled MLIR: https://github.com/IntelAI/mlir
\ No newline at end of file
+.. _precompiled MLIR: https://github.com/IntelAI/mlir
+
+Looking at graph objects
+------------------------
+
+A number of nGraph objects can print themselves on streams. For example,``cerr << a + b`` produces
+``v0::Add Add_2(Parameter_0[0]:f32{2,3}, Parameter_1[0]:f32{2,3}):(f32{2,3})`` indicating the
+specific version of the op, it's name, arguments, and outputs.
```
Comment (indie) [inline doc/sphinx/source/inspection/debug_core.rst]: its name, arguments, and outputs.

File: src/ngraph/node.cpp [hunk 2]
```diff
@@ -590,50 +590,35 @@ const op::AutoBroadcastSpec& Node::get_autob() const
...
-        sep = ", ";
+        out << "v" << get_type_info().version << "::" << get_type_info().name << " " << get_name()
+            << "(";
+        string sep = "";
+        for (auto arg : input_values())
+        {
+            out << sep << arg;
+            sep = ", ";
+        }
+        out << "):(";
```
Comment (rkimballn1) [inline src/ngraph/node.cpp]:
```suggestion
        out << ") -> (";
```

File: src/ngraph/node.hpp [hunk 3]
```diff
@@ -270,9 +270,11 @@ namespace ngraph
         virtual bool is_dynamic() const;
         virtual bool has_state() const { return false; }
         size_t get_instance_id() const { return m_instance_id; }
-        friend NGRAPH_API std::ostream& operator<<(std::ostream&, const Node&);
-        virtual std::ostream& write_short_description(std::ostream&) const;
-        virtual std::ostream& write_long_description(std::ostream&) const;
+        /// Writes a description of a node to a stream
```
Comment (rkimballn1) [inline src/ngraph/node.hpp]:
```suggestion
        /// \brief Writes a description of a node to a stream
```

File: src/ngraph/node.cpp (hunk 2 above)
Comment (rkimballn1) [inline src/ngraph/node.cpp]:
```suggestion
        out << ") -> (";
```

File: src/ngraph/type/element_type.cpp [hunk 4]
```diff
@@ -239,9 +239,12 @@ namespace ngraph
 
 std::ostream& element::operator<<(std::ostream& out, const element::Type& obj)
 {
-    out << "element::Type{" << obj.bitwidth() << ", " << obj.is_real() << ", " << obj.is_signed()
+    return out << obj.get_type_name();
+#if 0
```
Comment (rkimballn1) [inline src/ngraph/type/element_type.cpp]: Remove this block

**Pull request merged**